tagging/extend_tags.go

```diff
@@ -0,0 +1,137 @@
+package tagging
+
+import (
+	"sort"
+	"strings"
+)
+
+// ExtendTags holds the pre-calculated data to apply the
+// "extend tags" operation on a slice of strings representing
+// some tags
+type ExtendTags struct {
+	extraTags    []string
+	extraTagsMap map[string]string
+	dropPrefixes []string
+}
+
+// NewExtendTags creates a new ExtraTags struct, including
+// the pre-calculation of prefixes to drop. Empty tags are
```

Comment:
nit, can we call `dropPrefixes` something else? At the sink level we have `signalfx_metric_tag_prefix_drops` which refers to not the prefix defined here (i.e. the tag "key") but rather a string prefix on the key itself - this is confusing

Reply:
Good point!